### PR TITLE
feat(magnetModal): #MAG-401 add magnet duplication modal

### PR DESCRIPTION
**Pull request by alicedraillard**
## Issue ticket number and link
[MAG-401](https://jira.support-ent.fr/browse/MAG-401)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/components/board-create-magnet-magnet-modal/BoardCreateMagnetMagnetModal.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      cardIds: [],
+    }));
+  };
+
+  const onCloseModal = () => {
+    setInputValue(initialInputvalue);
+    onClose();
+  };
+
+  const createMagnetMagnet = async () => {
```
Comment (ThomasSevagen) [inline frontend/src/components/board-create-magnet-magnet-modal/BoardCreateMagnetMagnetModal.tsx:72]: nom de fonction : handleSubmit, toujours au submit

File: frontend/src/components/board-create-magnet-magnet-modal/types.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  open: boolean;
+  onClose: () => void;
+}
+
+export interface InputValueState {
+  search: string;
+  currentTab: CURRENTTAB_STATE;
+  isByBoards: boolean;
+  isByFavorite: boolean;
+  cardIds?: string[];
```
Comment (ThomasSevagen) [inline frontend/src/components/board-create-magnet-magnet-modal/types.ts:13]: pourquoi cards ids est optionnel?

File: frontend/src/components/board-create-magnet-magnet-modal/useRenderContent.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    myCardsResult?.all?.map((card: ICardItemResponse) =>
+      new Card().build(card),
+    ) || [];
+
+  const isOneMagnetInBoards = boardsWithCards.some(
+    (item: Board) => !!item.cards.length,
+  );
+
+  const updateSelectedMagnets = (cardId: string) => {
+    console.log("initial selection", inputValue.cardIds);
```
Comment (ThomasSevagen) [inline frontend/src/components/board-create-magnet-magnet-modal/useRenderContent.tsx:84]: log

File: frontend/src/components/board-create-magnet-magnet-modal/utils.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { InputValueState } from "./types";
+import { CURRENTTAB_STATE } from "../tab-list/types";
+
+export const initialInputvalue: InputValueState = {
```
Comment (ThomasSevagen) [inline frontend/src/components/board-create-magnet-magnet-modal/utils.ts:4]: renseigner cardsIds, peut être que c'est pour ça que tu l'avais mis en optionnel

File: frontend/src/components/board-create-magnet-magnet-modal/useRenderContent.tsx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    let updatedSelectedMagnets = inputValue.cardIds ?? [];
+    if (updatedSelectedMagnets.find((magnetId: string) => magnetId == cardId)) {
+      const index = updatedSelectedMagnets.indexOf(cardId, 0);
+      updatedSelectedMagnets.splice(index, 1);
+    } else {
+      updatedSelectedMagnets = [...updatedSelectedMagnets, cardId];
+    }
+    console.log({ updatedSelectedMagnets });
```
Comment (ThomasSevagen) [inline frontend/src/components/board-create-magnet-magnet-modal/useRenderContent.tsx:92]: log

**Review by niko-amosse — APPROVED** [review]

**Pull request merged**